server/svix-server/src/redis/mod.rs

```diff
@@ -134,28 +213,92 @@ impl<'a> ClusteredPooledConnection<'a> {
-    let mgr = RedisClusterConnectionManager::new(redis_dsn)
-        .expect("Error initializing redis cluster client");
-    let pool = bb8::Pool::builder()
-        .max_size(cfg.redis_pool_max_size.into())
-        .build(mgr)
-        .await
-        .expect("Error initializing redis cluster connection pool");
-    let pool = ClusteredRedisPool { pool };
-    RedisPool::Clustered(pool)
+pub struct ClusteredUnpooledConnection {
+    con: redis::cluster_async::ClusterConnection,
 }
 
-pub async fn new_redis_pool(redis_dsn: &str, cfg: &Configuration) -> RedisPool {
-    let mgr =
-        RedisMultiplexedConnectionManager::new(redis_dsn).expect("Error initializing redis client");
-    let pool = bb8::Pool::builder()
-        .max_size(cfg.redis_pool_max_size.into())
-        .build(mgr)
+impl ClusteredUnpooledConnection {
+    pub async fn query_async<T: FromRedisValue>(&mut self, cmd: redis::Cmd) -> RedisResult<T> {
+        cmd.query_async(&mut self.con).await
+    }
+
+    pub async fn query_async_pipeline<T: FromRedisValue>(
+        &mut self,
+        pipe: redis::Pipeline,
+    ) -> RedisResult<T> {
+        pipe.query_async(&mut self.con).await
+    }
+}
+
+async fn new_redis_pool_helper(
+    redis_dsn: &str,
+    clustered: bool,
+    max_connections: u16,
+) -> RedisPool {
+    if clustered {
+        let mgr = RedisClusterConnectionManager::new(redis_dsn)
+            .expect("Error initializing redis cluster client");
+        let pool = bb8::Pool::builder()
+            .max_size(max_connections.into())
+            .build(mgr)
+            .await
+            .expect("Error initializing redis cluster connection pool");
+        let pool = ClusteredRedisPool { pool };
+        RedisPool::Clustered(pool)
+    } else {
+        let mgr = RedisConnectionManager::new(redis_dsn).expect("Error initializing redis client");
+        let pool = bb8::Pool::builder()
+            .max_size(max_connections.into())
+            .build(mgr)
+            .await
+            .expect("Error initializing redis connection pool");
+        let pool = NonClusteredRedisPool { pool };
+        RedisPool::NonClustered(pool)
+    }
+}
+
+async fn new_redis_unpooled_helper(redis_dsn: &str, clustered: bool) -> RedisPool {
+    if clustered {
+        let cli = redis::cluster::ClusterClient::builder(vec![redis_dsn])
+            .retries(1)
+            .connection_timeout(REDIS_CONN_TIMEOUT)
+            .build()
+            .expect("Error initializing redis-unpooled cluster client");
+        let con = cli
+            .get_async_connection()
+            .await
+            .expect("Failed to get redis-cluster-unpooled connection");
+        RedisPool::ClusteredUnpooled(ClusteredRedisUnpooled { con })
+    } else {
+        let cli = redis::Client::open(redis_dsn).expect("Error initializing redis unpooled client");
+        let con = redis::aio::ConnectionManager::new_with_backoff_and_timeouts(
+            cli,
+            2,
```

Comment:
The values here match the library defaults, with exception of number of retries (1), which matches how we configure the clustered client.